sklearn/metrics/ranking.py

```diff
@@ -292,7 +299,23 @@ def _binary_roc_auc_score(y_true, y_score, sample_weight=None):
 
         fpr, tpr, tresholds = roc_curve(y_true, y_score,
                                         sample_weight=sample_weight)
-        return auc(fpr, tpr)
+        if max_fpr is None or max_fpr == 1:
+            return auc(fpr, tpr)
+
```

Comment:
WDYT about adding
```python
if max_fpr == 0:
    return 0
```
as the AUC is equal to 0 when max_fpr = 0

Reply:
Since setting ``max_fpr`` to zero makes no sense I think it might be best to raise a ValueError.